src/frontend/pages.ts

```diff
@@ -39,14 +40,14 @@ export const getRoute = <TParams, TReturn>(
   return path.join(parentPath, page.relativePath);
 };
 
-const getUrl = <TParams>(
+export const getUrl = <TParams>(
   page: Page<any, any>,
-  request: Request<TParams>
+  requestParams: TParams
 ): string => {
   const parentRoute = page.parent ? getRoute(page.parent) : "/";
   let url = path.join(parentRoute, page.relativePath);
   const regex = /:([A-Za-z_]+)/;
-  const params = request.params as Record<string, string>;
+  const params = requestParams as Record<string, string>;
   while (regex.exec(url)) {
     url = url.replace(regex, (a, b) => params[b]);
   }
```

Comment:
I do worry here about params[b] not existing -- we assume it's there but there's no checking.  This is an existing issue (that I introduced, but couldn't think of a better way) -- maybe we should throw an error at least instead of formatting an 'undefined' into the url silently?